server/src/main/java/org/elasticsearch/index/seqno/RetentionLeaseSyncAction.java

```diff
@@ -122,6 +122,12 @@ public void sync(
 
     @Override
     protected WritePrimaryResult<Request, Response> shardOperationOnPrimary(final Request request, final IndexShard primary) {
+        return performOnPrimary(request, primary, logger);
+    }
+
+    public static WritePrimaryResult<Request, Response> performOnPrimary(final Request request,
+                                                                         final IndexShard primary,
+                                                                         final Logger logger) {
```

Comment:
Can we assert that we hold an operation permit here?

Reply:
The sync action is too thin now, so I inline it directly to the test.